notation.go

```diff
@@ -49,8 +49,8 @@ type SignOptions struct {
 // and packing in various signature formats.
 type Signer interface {
 	// Sign signs the artifact described by its descriptor,
-	// and returns the signature and SignerInfo.
-	Sign(ctx context.Context, desc ocispec.Descriptor, opts SignOptions) ([]byte, *signature.SignerInfo, error)
+	// and returns the signature, annotations and SignerInfo.
+	Sign(ctx context.Context, desc ocispec.Descriptor, opts SignOptions) ([]byte, map[string]string, *signature.SignerInfo, error)
```

Comment:
Just an open discussion, another design option here is creating a new unexported interface, something like:
```
in notation.go:
    type signerAnnotation interface {
        PluginAnnotations(xxx) map[string]string
    }
    in func Sign(xxx) xxx {
         do a type assertion on the signer input, if it's a signerAnnotation, get its plugin annotations.
    }
in signer/plugin.go:
    type pluginSigner struct {
	plugin       plugin.SignPlugin
	keyID        string
	pluginConfig map[string]string
        pluginAnnotations map[string]string
    }
    // pluginSigner will implement notation.signerAnnotation
    func (s *pluginSigner) PluginAnnotations(xxx) map[string]string {
        return s.pluginAnnotations
    }
    inside func (s *pluginSigner) generateSignatureEnvelope(xxx) xxx {
         add logic: s.pluginAnnotations = resp.Annotations
    } 
```
In the above design, we are no longer breaking change the notation.Signer interface, and only pluginSigner is able to return annotations. (generic signer will not implement signerAnnotation as it doesn't generate annotation during the Sign process.) In the user's perspective, nothing really changes, the adding pluginAnnotations process happened under the hood.